src/jit/flowgraph.cpp

```diff
@@ -5455,15 +5455,15 @@ unsigned Compiler::fgMakeBasicBlocks(const BYTE* codeAddr, IL_OFFSET codeSize, B
                     if (!impIsTailCallILPattern(tailCall, opcode, codeAddr + sz, codeEndp, isRecursive,
                                                 &isCallPopAndRet))
                     {
-#ifdef _TARGET_AMD64_
+#ifndef FEATURE_CORECLR
```

Comment:
You need to retain the _TARGET_AMD64_ condition as well.  Even on the (full) .NET Framework, we wouldn't want to infect other architectures with this behavior.